### PR TITLE
Fix python-daemon install bug

### LINKED ISSUE #288: Kytos setup.py install is probably broken
Did the following steps:
1. Created a new virtual environment
2. Installed python-openflow from the repo, using setup.py install
3. Installed kytos-utils from the repo, using setup.py install.
4. Cloned kytos from the repo, and ran setup.py install.
In step 4, I got the following error (probably regarding the python-daemon installation):

```
(bbb) renn@renn-HP:~/kytest/kytos$ python setup.py install
running install
running bdist_egg
running egg_info
writing kytos.egg-info/PKG-INFO
writing dependency_links to kytos.egg-info/dependency_links.txt
writing requirements to kytos.egg-info/requires.txt
writing top-level names to kytos.egg-info/top_level.txt
reading manifest file 'kytos.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'kytos.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-x86_64/egg
running install_lib
running build_py
creating build/bdist.linux-x86_64/egg
creating build/bdist.linux-x86_64/egg/kytos
creating build/bdist.linux-x86_64/egg/kytos/core
copying build/lib/kytos/core/exceptions.py -> build/bdist.linux-x86_64/egg/kytos/core
copying build/lib/kytos/core/constants.py -> build/bdist.linux-x86_64/egg/kytos/core
copying build/lib/kytos/core/api_server.py -> build/bdist.linux-x86_64/egg/kytos/core
copying build/lib/kytos/core/daemon.py -> build/bdist.linux-x86_64/egg/kytos/core
copying build/lib/kytos/core/websocket.py -> build/bdist.linux-x86_64/egg/kytos/core
copying build/lib/kytos/core/config.py -> build/bdist.linux-x86_64/egg/kytos/core
copying build/lib/kytos/core/events.py -> build/bdist.linux-x86_64/egg/kytos/core
copying build/lib/kytos/core/flow.py -> build/bdist.linux-x86_64/egg/kytos/core
copying build/lib/kytos/core/core.py -> build/bdist.linux-x86_64/egg/kytos/core
copying build/lib/kytos/core/__init__.py -> build/bdist.linux-x86_64/egg/kytos/core
copying build/lib/kytos/core/buffers.py -> build/bdist.linux-x86_64/egg/kytos/core
copying build/lib/kytos/core/napps_manager.py -> build/bdist.linux-x86_64/egg/kytos/core
copying build/lib/kytos/core/switch.py -> build/bdist.linux-x86_64/egg/kytos/core
copying build/lib/kytos/core/helpers.py -> build/bdist.linux-x86_64/egg/kytos/core
copying build/lib/kytos/core/napps.py -> build/bdist.linux-x86_64/egg/kytos/core
copying build/lib/kytos/core/tcp_server.py -> build/bdist.linux-x86_64/egg/kytos/core
copying build/lib/kytos/__init__.py -> build/bdist.linux-x86_64/egg/kytos
creating build/bdist.linux-x86_64/egg/tests
creating build/bdist.linux-x86_64/egg/tests/test_core
copying build/lib/tests/test_core/test_tcp_server.py -> build/bdist.linux-x86_64/egg/tests/test_core
copying build/lib/tests/test_core/__init__.py -> build/bdist.linux-x86_64/egg/tests/test_core
copying build/lib/tests/test_core/test_interface.py -> build/bdist.linux-x86_64/egg/tests/test_core
byte-compiling build/bdist.linux-x86_64/egg/kytos/core/exceptions.py to exceptions.cpython-36.pyc
byte-compiling build/bdist.linux-x86_64/egg/kytos/core/constants.py to constants.cpython-36.pyc
byte-compiling build/bdist.linux-x86_64/egg/kytos/core/api_server.py to api_server.cpython-36.pyc
byte-compiling build/bdist.linux-x86_64/egg/kytos/core/daemon.py to daemon.cpython-36.pyc
byte-compiling build/bdist.linux-x86_64/egg/kytos/core/websocket.py to websocket.cpython-36.pyc
byte-compiling build/bdist.linux-x86_64/egg/kytos/core/config.py to config.cpython-36.pyc
byte-compiling build/bdist.linux-x86_64/egg/kytos/core/events.py to events.cpython-36.pyc
byte-compiling build/bdist.linux-x86_64/egg/kytos/core/flow.py to flow.cpython-36.pyc
byte-compiling build/bdist.linux-x86_64/egg/kytos/core/core.py to core.cpython-36.pyc
byte-compiling build/bdist.linux-x86_64/egg/kytos/core/__init__.py to __init__.cpython-36.pyc
byte-compiling build/bdist.linux-x86_64/egg/kytos/core/buffers.py to buffers.cpython-36.pyc
byte-compiling build/bdist.linux-x86_64/egg/kytos/core/napps_manager.py to napps_manager.cpython-36.pyc
byte-compiling build/bdist.linux-x86_64/egg/kytos/core/switch.py to switch.cpython-36.pyc
byte-compiling build/bdist.linux-x86_64/egg/kytos/core/helpers.py to helpers.cpython-36.pyc
byte-compiling build/bdist.linux-x86_64/egg/kytos/core/napps.py to napps.cpython-36.pyc
byte-compiling build/bdist.linux-x86_64/egg/kytos/core/tcp_server.py to tcp_server.cpython-36.pyc
byte-compiling build/bdist.linux-x86_64/egg/kytos/__init__.py to __init__.cpython-36.pyc
byte-compiling build/bdist.linux-x86_64/egg/tests/test_core/test_tcp_server.py to test_tcp_server.cpython-36.pyc
byte-compiling build/bdist.linux-x86_64/egg/tests/test_core/__init__.py to __init__.cpython-36.pyc
byte-compiling build/bdist.linux-x86_64/egg/tests/test_core/test_interface.py to test_interface.cpython-36.pyc
installing package data to build/bdist.linux-x86_64/egg
running install_data
creating build/bdist.linux-x86_64/egg/EGG-INFO
installing scripts to build/bdist.linux-x86_64/egg/EGG-INFO/scripts
running install_scripts
running build_scripts
creating build/bdist.linux-x86_64/egg/EGG-INFO/scripts
copying build/scripts-3.6/kytosd -> build/bdist.linux-x86_64/egg/EGG-INFO/scripts
changing mode of build/bdist.linux-x86_64/egg/EGG-INFO/scripts/kytosd to 775
copying kytos.egg-info/PKG-INFO -> build/bdist.linux-x86_64/egg/EGG-INFO
copying kytos.egg-info/SOURCES.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying kytos.egg-info/dependency_links.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying kytos.egg-info/not-zip-safe -> build/bdist.linux-x86_64/egg/EGG-INFO
copying kytos.egg-info/requires.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying kytos.egg-info/top_level.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
creating 'dist/kytos-2017.1b1-py3.6.egg' and adding 'build/bdist.linux-x86_64/egg' to it
removing 'build/bdist.linux-x86_64/egg' (and everything under it)
Processing kytos-2017.1b1-py3.6.egg
removing '/home/renn/kytest/bbb/lib/python3.6/site-packages/kytos-2017.1b1-py3.6.egg' (and everything under it)
creating /home/renn/kytest/bbb/lib/python3.6/site-packages/kytos-2017.1b1-py3.6.egg
Extracting kytos-2017.1b1-py3.6.egg to /home/renn/kytest/bbb/lib/python3.6/site-packages
kytos 2017.1b1 is already the active version in easy-install.pth
Installing kytosd script to /home/renn/kytest/bbb/bin

Installed /home/renn/kytest/bbb/lib/python3.6/site-packages/kytos-2017.1b1-py3.6.egg
Processing dependencies for kytos==2017.1b1
Searching for python-daemon>=2.1.2
Reading https://pypi.python.org/simple/python-daemon/
Best match: python-daemon 2.1.2
Downloading https://pypi.python.org/packages/b2/fb/a280d65f81e9d69989c8d6c4e0bb18d7280cdcd6d406a2cc3f4eb47d4402/python-daemon-2.1.2.tar.gz#md5=9c57343d81f2a96c51cffeab982b04d2
Processing python-daemon-2.1.2.tar.gz
Writing /tmp/easy_install-yux7c4ac/python-daemon-2.1.2/setup.cfg
Running python-daemon-2.1.2/setup.py -q bdist_egg --dist-dir /tmp/easy_install-yux7c4ac/python-daemon-2.1.2/egg-dist-tmp-30wmhbv6
Traceback (most recent call last):
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/dist.py", line 353, in fetch_build_egg
    cmd = self._egg_fetcher
AttributeError: 'ChangelogAwareDistribution' object has no attribute '_egg_fetcher'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/sandbox.py", line 154, in save_modules
    yield saved
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/sandbox.py", line 195, in setup_context
    yield
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/sandbox.py", line 243, in run_setup
    DirectorySandbox(setup_dir).run(runner)
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/sandbox.py", line 273, in run
    return func()
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/sandbox.py", line 242, in runner
    _execfile(setup_script, ns)
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/sandbox.py", line 46, in _execfile
    exec(code, globals, locals)
  File "/tmp/easy_install-yux7c4ac/python-daemon-2.1.2/setup.py", line 86, in <module>
    
  File "/usr/lib/python3.6/distutils/core.py", line 108, in setup
    _setup_distribution = dist = klass(attrs)
  File "/tmp/easy_install-yux7c4ac/python-daemon-2.1.2/version.py", line 640, in __init__
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/dist.py", line 269, in __init__
    self.fetch_build_eggs(attrs['setup_requires'])
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/dist.py", line 313, in fetch_build_eggs
    replace_conflicting=True,
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/pkg_resources/__init__.py", line 826, in resolve
    dist = best[req.key] = env.best_match(req, ws, installer)
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/pkg_resources/__init__.py", line 1092, in best_match
    return self.obtain(req, installer)
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/pkg_resources/__init__.py", line 1104, in obtain
    return installer(requirement)
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/dist.py", line 378, in fetch_build_egg
    cmd.ensure_finalized()
  File "/usr/lib/python3.6/distutils/cmd.py", line 107, in ensure_finalized
    self.finalize_options()
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/command/easy_install.py", line 246, in finalize_options
    'dist_version': self.distribution.get_version(),
  File "/tmp/easy_install-yux7c4ac/python-daemon-2.1.2/version.py", line 656, in get_version
  File "/tmp/easy_install-yux7c4ac/python-daemon-2.1.2/version.py", line 651, in get_version_info
  File "/tmp/easy_install-yux7c4ac/python-daemon-2.1.2/version.py", line 552, in get_changelog_path
  File "/usr/lib/python3.6/posixpath.py", line 154, in dirname
    p = os.fspath(p)
TypeError: expected str, bytes or os.PathLike object, not NoneType

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "setup.py", line 153, in <module>
    zip_safe=False)
  File "/usr/lib/python3.6/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/lib/python3.6/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/command/install.py", line 67, in run
    self.do_egg_install()
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/command/install.py", line 117, in do_egg_install
    cmd.run()
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/command/easy_install.py", line 414, in run
    self.easy_install(spec, not self.no_deps)
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/command/easy_install.py", line 644, in easy_install
    return self.install_item(None, spec, tmpdir, deps, True)
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/command/easy_install.py", line 695, in install_item
    self.process_distribution(spec, dist, deps)
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/command/easy_install.py", line 740, in process_distribution
    [requirement], self.local_index, self.easy_install
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/pkg_resources/__init__.py", line 826, in resolve
    dist = best[req.key] = env.best_match(req, ws, installer)
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/pkg_resources/__init__.py", line 1092, in best_match
    return self.obtain(req, installer)
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/pkg_resources/__init__.py", line 1104, in obtain
    return installer(requirement)
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/command/easy_install.py", line 663, in easy_install
    return self.install_item(spec, dist.location, tmpdir, deps)
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/command/easy_install.py", line 693, in install_item
    dists = self.install_eggs(spec, download, tmpdir)
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/command/easy_install.py", line 873, in install_eggs
    return self.build_and_install(setup_script, setup_base)
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/command/easy_install.py", line 1101, in build_and_install
    self.run_setup(setup_script, setup_base, args)
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/command/easy_install.py", line 1087, in run_setup
    run_setup(setup_script, args)
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/sandbox.py", line 246, in run_setup
    raise
  File "/usr/lib/python3.6/contextlib.py", line 100, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/sandbox.py", line 195, in setup_context
    yield
  File "/usr/lib/python3.6/contextlib.py", line 100, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/sandbox.py", line 166, in save_modules
    saved_exc.resume()
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/sandbox.py", line 141, in resume
    six.reraise(type, exc, self._tb)
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/pkg_resources/_vendor/six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/sandbox.py", line 154, in save_modules
    yield saved
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/sandbox.py", line 195, in setup_context
    yield
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/sandbox.py", line 243, in run_setup
    DirectorySandbox(setup_dir).run(runner)
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/sandbox.py", line 273, in run
    return func()
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/sandbox.py", line 242, in runner
    _execfile(setup_script, ns)
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/sandbox.py", line 46, in _execfile
    exec(code, globals, locals)
  File "/tmp/easy_install-yux7c4ac/python-daemon-2.1.2/setup.py", line 86, in <module>
    
  File "/usr/lib/python3.6/distutils/core.py", line 108, in setup
    _setup_distribution = dist = klass(attrs)
  File "/tmp/easy_install-yux7c4ac/python-daemon-2.1.2/version.py", line 640, in __init__
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/dist.py", line 269, in __init__
    self.fetch_build_eggs(attrs['setup_requires'])
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/dist.py", line 313, in fetch_build_eggs
    replace_conflicting=True,
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/pkg_resources/__init__.py", line 826, in resolve
    dist = best[req.key] = env.best_match(req, ws, installer)
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/pkg_resources/__init__.py", line 1092, in best_match
    return self.obtain(req, installer)
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/pkg_resources/__init__.py", line 1104, in obtain
    return installer(requirement)
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/dist.py", line 378, in fetch_build_egg
    cmd.ensure_finalized()
  File "/usr/lib/python3.6/distutils/cmd.py", line 107, in ensure_finalized
    self.finalize_options()
  File "/home/renn/kytest/bbb/lib/python3.6/site-packages/setuptools/command/easy_install.py", line 246, in finalize_options
    'dist_version': self.distribution.get_version(),
  File "/tmp/easy_install-yux7c4ac/python-daemon-2.1.2/version.py", line 656, in get_version
  File "/tmp/easy_install-yux7c4ac/python-daemon-2.1.2/version.py", line 651, in get_version_info
  File "/tmp/easy_install-yux7c4ac/python-daemon-2.1.2/version.py", line 552, in get_changelog_path
  File "/usr/lib/python3.6/posixpath.py", line 154, in dirname
    p = os.fspath(p)
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

### PR DESCRIPTION
**Pull request by cemsbr**
This is an upstream bug that was solved by forking it.
Fix #288

### CODE REVIEW

**Review by beraldoleal — COMMENTED** [review]
Do you forked ? Do you really think that this is a good idea? Who is going to maintain this updated ?

**Comment by cemsbr** [comment]
Yes, I forked and I think it was a good idea because upstream has had this issue unfixed for 4 months. When upstream releases the next version, it will be used automatically because our requirement is ">=2.1.2".

**Comment by beraldoleal** [comment]
@cemsbr did you submitted a PR to upstream too ? I think that this is important.

**Comment by beraldoleal** [comment]
I'm going to accept. But we have to pay attention to this in order to revert this on near future.

File: setup.py [hunk 1]
```diff
@@ -141,6 +141,9 @@ def create_path(self, file_name):
       license='MIT',
       test_suite='tests',
       install_requires=requirements,
+      dependency_links=[
+          'https://github.com/cemsbr/python-daemon/tarball/latest_release#egg=python-daemon-2.1.2'
```
Comment (diraol) [inline setup.py]: Please, add `#noqa: E501` at the end of line 145.

**Review by diraol — CHANGES_REQUESTED** [review]
@cemsbr, run python setup.py test before re-sending this PR, I think that there are some other [isort] errors. At least, after adding the `#noqa` here locally, the tests passed on this URL line but they broke on other files because of isort errors.

**Comment by cemsbr** [comment]
@beraldoleal , yes, I did it in [their issue tracker](https://pagure.io/python-daemon/issue/2). I'm also watching the project to know when a new version is released.

File: setup.py (hunk 1 above)
Comment (cemsbr) [reply setup.py]: Thanks, @diraol. I split the string so I don't need to ignore checks.

**Review by diraol — APPROVED** [review]

**Comment by beraldoleal** [comment]
@cemsbr cool, thanks for that.

**Pull request merged**